> 19-May-2025

> MS: MBE-25-0312
> Title: Accessible, realistic genome simulation with selection using stdpopsim
> 
> Dear Dr. Kern:
> 
> The in-depth review of your manuscript by the editors and the peer reviewers is now complete. Based on their assessment, it is clear that your manuscript requires a substantial revision before it can be considered further for publication in MBE. Comments from the editors and external reviewers are included below.
> 
> We invite you to revise your manuscript within 60 days and submit it for further consideration. If you need an extension, please contact Chloe Masten by e-mail ([EAssist.MBE@gmail.com](mailto:EAssist.MBE@gmail.com)) before the deadline. If your revision option expires and you still intend to submit a revision, please also contact the editorial office.
> 
> Please note that all articles accepted for publication will be published as Open Access and subject to payment of an Article Processing Charge (APC). Find out more on our Open Access page here:  [https://academic.oup.com/mbe/pages/Open_Access](https://urldefense.com/v3/__https://academic.oup.com/mbe/pages/Open_Access__;!!C5qS4YX3!BQwWqW_BDtRbdEKUIOE45D2E5VpCb6xERj2wa5kbf04aQtd0YFhwy6TX7GRKG3Qnu6_Hg0xpxPoE5xJARQA1SyCM-MUA$)
> 
> If you are planning to promote your manuscript in the press, please contact the Editorial Office ([eassist.mbe@publishingsolutions.net](mailto:eassist.mbe@publishingsolutions.net))  as soon as possible so we can take appropriate action.
> 
> Most importantly, the revised manuscript will be subject to editorial and external reviews, and its eventual acceptance depends on the reviewers’ and editors' enthusiasm. Note that manuscripts invited to be revised are accepted at a very high rate. But, manuscripts deemed to require more than one major revision are often rejected. So, you must revise it to satisfy all editorial and reviewer concerns. A re-review by original and new reviewers may raise additional concerns, so anticipate them in advance and change thoroughly.
> 
> You can create a revision by using the URL: *** PLEASE NOTE: This is a two-step process. After clicking on the link, you will be directed to a webpage to confirm. ***
> 
> *** PLEASE NOTE: This is a two-step process. After clicking on the link, you will be directed to a webpage to confirm. ***
> 
> [https://mc.manuscriptcentral.com/mbe?URL_MASK=688f965e55744d13a78b527a4fa47659](https://urldefense.com/v3/__https://mc.manuscriptcentral.com/mbe?URL_MASK=688f965e55744d13a78b527a4fa47659__;!!C5qS4YX3!BQwWqW_BDtRbdEKUIOE45D2E5VpCb6xERj2wa5kbf04aQtd0YFhwy6TX7GRKG3Qnu6_Hg0xpxPoE5xJARQA1Szf2KD5E$)
> 
> Following is a checklist of needed actions and files.
> 
> [#1]  Number and respond to each of the comments of the reviewers and the Editor. You must enter these comments in the Authors' Response section on the website when you submit the revision.
> 
> [#2]  Please follow the Author Guidelines at: ([https://academic.oup.com/mbe/pages/General_Instructions](https://urldefense.com/v3/__https://academic.oup.com/mbe/pages/General_Instructions__;!!C5qS4YX3!BQwWqW_BDtRbdEKUIOE45D2E5VpCb6xERj2wa5kbf04aQtd0YFhwy6TX7GRKG3Qnu6_Hg0xpxPoE5xJARQA1S9eAZ3qU$)
> In the File Upload section of the submission website, please provide the following:
> 
> [#3]  Upload a file in DOC/DOCX format, RTF format, or LaTeX format containing the text, tables and figure legends. If you are submitting a LaTex/Tex file, please include accompanying .tex files (e.g., .bib, .sty, .cls,.bst).
> 
> [#4]  Upload files containing high-resolution figures – this does not apply to any supplementary figures. If you have color figures, please confirm which ones are to be published in color in print. If you have color figures that are to be published color online only, please ensure the figure legends do not refer to color.  Images are required as high-resolution files (600-1200 dpi for line drawings and 350 dpi for color and half-tone artwork). Most formats are acceptable, including eps, tiff, pdf, jpg, and Powerpoint.
> 
> [#5]  Upload a copy of the manuscript in PDF (for “Advance Access”) that contains the complete manuscript, including text, any tables, figure legends, and figures, but excluding any supplementary material.
> 
> [#6]  Upload any supplementary information files. There is no limit on supplementary files.  All supplementary information, including supplementary tables and supplementary figure legends, must be uploaded separately as "supplementary files" and not placed in the manuscript. ([https://academic.oup.com/mbe/pages/Supplementary_Information](https://urldefense.com/v3/__https://academic.oup.com/mbe/pages/Supplementary_Information__;!!C5qS4YX3!BQwWqW_BDtRbdEKUIOE45D2E5VpCb6xERj2wa5kbf04aQtd0YFhwy6TX7GRKG3Qnu6_Hg0xpxPoE5xJARQA1Sy3yzMtw$)).
> 
> [#7]  Double-check the format of citations and references to ensure that they meet MBE requirements ([https://academic.oup.com/mbe/pages/General_Author_Guidelines#References](https://urldefense.com/v3/__https://academic.oup.com/mbe/pages/General_Author_Guidelines*References__;Iw!!C5qS4YX3!BQwWqW_BDtRbdEKUIOE45D2E5VpCb6xERj2wa5kbf04aQtd0YFhwy6TX7GRKG3Qnu6_Hg0xpxPoE5xJARQA1S_MtXt15$)). Failure to do so will result in a delay in manuscript processing.
> 
> [#8]  Double-check the format for revisions.  Articles, Letters, Brief Communications, Perspectives, Reviews, and Protocols all have different format requirements.
> Articles must have headings in order: Introduction, Results, Discussion, and Materials and Methods. Results and Discussion may be combined. Subsections are allowed throughout.
> Brief Communications must be in the resources category.
>   ([https://academic.oup.com/mbe/pages/manuscript_types](https://urldefense.com/v3/__https://academic.oup.com/mbe/pages/manuscript_types__;!!C5qS4YX3!BQwWqW_BDtRbdEKUIOE45D2E5VpCb6xERj2wa5kbf04aQtd0YFhwy6TX7GRKG3Qnu6_Hg0xpxPoE5xJARQA1S88fMR6p$))
> 
> [#9] Your original files will be available to you when you upload your revised manuscript. Please delete any redundant files before completing the submission.
> 
> We look forward to receiving a revised version of the manuscript for further consideration.
> 
> Sincerely,
> 
> Board of Editors
> Molecular Biology and Evolution
> 
> Associate Editor
> Editors’ comments to the author:
> Two experts in population genetics examined this manuscript and gave Very High and Medium marks in the Impact of the resource reported. Despite this mixed evaluations, I think this extension of stdpopsim contributed by many authors should be very useful in the community of population genomics. However, referees found many places in the manuscript that need clarification. These comments should be addressed in revision. In addition, regarding the point discussed in lines 368-373, a recent paper Marsh & Johri 2024 MBE 41(7):msae118 might be acknowledged because it also developed simulations including the annonation for D. mel and humans and also mutation and recombination rate heterogeneity.
> 
> Reviewers' comments:
> Reviewer: 1
> 
> Comments to the Author
> In this manuscript, a large team of authors (Gower, Pope, Rodrigues, Tittes et al) present the next set of developments for the stdpopsim software, which facilitates "standard" population-genetic simulations that can be set to follow estimated demographic and genetic (mutation rates, recombination maps) parameters in a number of species and populations, as well as a framework for entering new species and populations into the collection. stdpopsim has become an important tool in population-genetic research. The authors here represent a large community of people invested in the maintenance and improvement of the software, which is open and undergoes thorough unit testing. The addition of selection to stdpopsim's capabilities is a substantial advance warranting a new publication. MBE also seems to me an appropriate outlet.
> 
> In addition to announcing the new features available in stdpopsim, the paper describes three applications of the software: one on demographic inference in the presence of background selection, another on DFE estimation, and a third on detection of selective sweeps. Two of these contrast the results obtained using human vs. vaquita porpoise parameters.
> 
> I am generally supportive of the paper and of the broader stdpopsim project. That said, I do have some constructive thoughts and questions:
> 
> ---Perhaps my main constructive reaction is that this paper is a bit light on information about the ways in which the new selection features are implemented. There is a section "implementing selection in stdpopsim", but it is not always easy to figure out from the description here how things work under the hood. Instead, the section is more like a short tour of the new features aimed at users. Moreover, there is no section of the methods that really goes over the details of implementation. The ms says that SLiM is the backend, but there aren't many details about how information passed into stdpopsim is translated into SLiM. It's also not clear to me how the selective sweep mode is implemented, given that it gives users the option to condition on some range of starting and ending allele frequencies. I assume that the approach is to run full SLiM iterations until the sweep ends up as desired (though it may not be)? But one might also simulate the allele-frequency trajectory for the sweeping allele alone until one "works", and then do coalescent simulations to fill in flanking variation conditional on the allele-frequency trajectory, a la Hudson & Kaplan '88 (though I don't quite see how one would do this simultaneously with background selection, which the sweep simulations include). In any case, I'm not sure how the conditions are being achieved from reading the paper. (I've refrained from looking at the online documentation for stdpopsim 0.3.0, where there may be more answers, in order to review the paper as an independent product.) I don't think the paper necessarily needs loads more text on this, but I think there should be more information hinting at implementation in main text and probably a new methods subsection with further details.
> 
> ---Relatedly, why is the dominance coefficient currently limited to between 0 and 1? It's fine that it is for now, but I was curious what makes it more difficult to include other values of h.
> 
> ---I think it would be helpful to say a bit more about the vaquita earlier in the manuscript. Although some of the authors have worked on the vaquita recently, it is still a fairly obscure organism, so it would be worth describing its unfortunate situation and why it is an interesting comparison with humans throughout these simulations.
> 
> ---It seems as though everything on the backend is being done by SLiM in the demonstrations here (unless I'm missing something about how the sweeps are done), so the runtimes are not really the product of the effort of the authors here. Still, I think it would be good to provide a bit more information on the runtimes for prospective users to get a sense of what doing selection in stdpopsim entails. (It's mentioned that the sweeps were time-consuming, but details aren't given as far as I can tell.)
> 
> ---Point of clarification about the flexibility of DFE specification: Is the DFE allowed to be a mixture distribution in a single genomic region? Generally, the paper mentions users can provide a custom DFE, but what does that mean in practice? Does it have to be of a particular distribution family or form? (i.e. the paper mentions "fixed value, normally-distributed, gamma-distributed"). The vaquita DFE seems to be a mixture of several nonoverlapping DFEs with different h values, which suggests to me substantial flexibility, but I'm not quite clear on the limits. Most of the figures feature fairly few replicates, which makes me think that these simulations are necessarily time/compute intensive.
> 
> ---Sentence starting on line 383 about difficulty identifying h and s separately seems to be missing some important citations, e.g. Fuller et al 2019 (pmid 30962618) and citations therein, including Simons et al 2014; Wright 1937.
> 
> ---It might bear further comment that the gamma distribution is not flexible enough to accommodate the distribution of hs in the vaquita simulation (which is what the DFE inference methods can really see), even if the s values are gamma distributed (bc of the big jump caused in hs by the switch in dominance). This is interesting given how often gammas are used in DFE inference. The paper does mention that using h=1/2 as an assumption can lead to bad estimation of s, but the point that a relationship between s and h can cause hs to be non-gamma even when s is gamma is perhaps a little more subtle. Generally, I thought the lessons learned from the DFE exercise here could be put more clearly.
> 
> ---Related to the previous point, I found it a bit odd to that both the style of the DFE (and in particular the h/s relationship) and the demography switched when comparing vaquita and human DFE. It seemed like a missed opportunity to do something hinted at earlier in the manuscript: namely, to try pairing demographies and DFEs from different species. I.e. it would be interesting in this context to look at simulations that have the human demography and vaquita DFE and vice versa, even if only in a supplementary figure.
> 
> ---Figure S1, right panel. Is pi being computed from all the genomes (CEU/CHB/YRI)? I think it's okay given the point that's being made, but it might strike some readers as odd given how much is often made of the pi differences in these groups under the estimated demography, so maybe it should be said explicitly what's being done.
> 
> ---Last comment, and this may be an idiosyncratic opinion on my part, so I do not expect the authors to implement, respond to, or even quote this point in the response letter. I'm merely offering a reaction. In my opinion, three motifs are at a slightly awkward level of detail/coverage. On one hand, they're not enough to please partisans or enthusiasts of any of these three areas. (The paper is clear about the fact that they're not meant to be thorough benchmarks.) On the other hand, they are a big part of the paper, substantial on their own, and the bulk of the discussion is given over to their substantive results rather than to stdpopsim per se. I personally find many of the results interesting, but I also wondered whether the paper might be stronger if these segments were streamlined. One way to do this might be to focus less on the three topics themselves, but to change the emphasis to how selection results change when the species changes, i.e. to focus the thread of the results primarily on a quick inspection of the human/vaquita differences (one might imagine adding in Drosophila too in such a framing) rather than on separate questions about the effect of BGS on demographic estimation and sweep inference, and on the effect of h-s relationship on DFE inference. Again, this is merely offered in the spirit of a friendly comment, and I expect no response.
> 
> Reviewer: 2
> 
> Comments to the Author
> In this manuscript, Gower et al. extend the functionality of stdpopsim to include selection. They test the performance of a few different methods that infer historical population sizes, the distribution of fitness effects of new deleterious mutations, and those that detect selective sweeps using the simulations in SLiM. The added functionality will be useful to the users of stdpopsim. Below are some specific comments:
> 
> 1) How do you deal with overlapping annotations in a gff?
> 2) The inference of the deleterious DFE with GRAPES is quite inaccurate. Is it possible that there is a factor of 2 error involved?
> 3) Figure 3c does not have any error bars, it’s difficult to conclude if the inference is actually significantly different from the truth. Which “N” was used to obtain estimates of “s” in Figure 3a? This should be mentioned in the legend.
> 
> Minor comments:
> 1)You cite only Gillespie 1991 on line 56. Requires many more citations, with at least one to Kimura.
> 2)What fraction of the genome is under selection in the human genome you simulated?
> 3)”This is somewhat surprising given that background selection is expected to somewhat interfere with the sweep signature.” (line 349-350) -> I’m not sure what this means. It’s unclear to me why background selection will interfere with a sweep signature. Do you have a citation for this?